Hakawai/Mentions/HKWMentionsStartDetectionStateMachine.m

```diff
@@ -298,6 +330,20 @@ - (void)mentionCreationResumed {
     self.state = HKWMentionsStartDetectionStateCreatingMention;
 }
 
+#pragma mark - Public helper method
+
++ (NSString *)wordAfterLocation:(NSUInteger)location text:(NSString *)text {
+    NSString *charString;
+    NSMutableString *const word = [[NSMutableString alloc] init];
+    for(NSUInteger i = location; i < text.length ; i++) {
+        charString =  [text substringWithRange:NSMakeRange(i, 1)];
+        if([text isEqualToString:@" "]) {
```

Comment:
what does this do

Reply:
When a control character ("@" or "#") is inserted before an already existing word, we would need that word, to begin/trigger a mention with or replace the word with the selected mention. This method returns the characters after a certain location till the next delimiter (whitespace/newline) as a word
Updated logic to use `whitespaceAndNewLine` character set instead hard code checks.